### PR TITLE
Initial quality system docs

### PR DESCRIPTION
**Pull request by AMZN-alexpete**
## Change summary

Initial quality system docs which cover the functionality available in stabilization.  
Related RFC:  https://github.com/o3de/sig-core/issues/57

### Submission Checklist:

* [x] **Descriptive active voice** - Do descriptive sentences have a clear *subject* and *action verb*?
* [x] **Answer the question at hand** - Does the documentation answer a *what*, *why*, *how*, or *where* type of question?
* [x] **Consistency** - Does the content consistently follow the [Style Guide](https://o3de.org/docs/contributing/to-docs/style-guide/quick-reference)?
* [x] **Help the user** - Does the documentation show the user something *meaningful*?

### CODE REVIEW

File: content/docs/user-guide/testing/quality/_index.md [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+
+### Project/Registry/Platform/Android/quality.setreg
+
+```json
+{
+    "O3DE": {
+        "Quality": {
+            "Groups": {
+                "q_general": {
+                    "Default": 0 // change the default level to use to be Low
```
Comment (tkothadev) [inline content/docs/user-guide/testing/quality/_index.md:206]: nit:
`// change the default level to use to be Low`
to
`// change the default level to Low`

File: content/docs/user-guide/testing/quality/_index.md [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+
+### Project/Registry/quality.setreg
+
+```json
+{
+    "O3DE": {
+        "Quality": {
+            "Groups": {
+                "q_general": {
+                    "Default": 3 // change the default level to use to be VeryHigh instead of High
```
Comment (tkothadev) [inline content/docs/user-guide/testing/quality/_index.md:177]: nit:
`// change the default level to use to be VeryHigh instead of High`
to
`// change the default level to VeryHigh instead of High`

**Review by tkothadev — APPROVED** [review]

File: content/docs/user-guide/testing/_index.md [hunk 3]
```diff
@@ -4,7 +4,7 @@ date: 2021-03-02T00:23:56-05:00
 weight: 1300
 ---
 
-Test automation is the process of writing small snippets of code (tests) to verify the desired functionality of Open 3D Engine extensions and tools that you create.
+Test automation is the process of writing small snippets of code (tests) to verify the desired functionality of Open 3D Engine extensions and tools that you create.  Optimization is the process of making code perform and use resources in the most efficient way.
```
Comment (lumberyard-employee-dm) [inline content/docs/user-guide/testing/_index.md:7]: What does the optimization blurb have to do with testing?
It doesn't seem relevant to this document

File: content/docs/user-guide/testing/quality/_index.md [hunk 4]
```diff
@@ -0,0 +1,216 @@
+---
+description: Use the Quality system to define quality levels and the console variable settings for each level.
+title: Quality System
+linkTitle: Quality System
+---
```
Comment (lumberyard-employee-dm) [inline content/docs/user-guide/testing/quality/_index.md:5]: This document seems like it should either be under the existing "visualization" directory or added to a new "quality" or "performance" directory.

I wouldn't consider this directly related to testing.
The impact is more on having the quality level set for the end user to get optimal performance on their device.

File: content/docs/user-guide/testing/_index.md (hunk 3 above)
Comment (AMZN-alexpete) [reply content/docs/user-guide/testing/_index.md:7]: The high level title/link to this document is `Testing and optimization` so I added a blurb about optimization - but I agree, it does seem like maybe I should just remove "optimization" from the title and make a new section entirely for optimization

Comment (lumberyard-employee-dm) [reply content/docs/user-guide/testing/_index.md:7]: "pptimization" sounds a good folder to place the settings into

File: content/docs/user-guide/testing/quality/_index.md [hunk 5]
```diff
@@ -0,0 +1,216 @@
...
+            "Groups": {
+                "q_general": {
+                    "Description" : "Description with values e.g.. 0 : Low, 1 : Medium, 2 : High",
+                    "Levels": [ // quality level names
+                        "Low",
+                        "Medium",
+                        "High",
+                        ...
+                    ],
+                    "Default": 3, // default quality level index in the Levels array
```
Comment (lumberyard-employee-dm) [inline content/docs/user-guide/testing/quality/_index.md:34]: We should probably change this example quality level settings Default value to be an index that is within this group range of 0-2, since this description only goes up to "HIgh".
```suggestion
                    "Description" : "Description with values e.g.. 0 : Low, 1 : Medium, 2 : High",
                    "Levels": [ // quality level names
                        "Low",
                        "Medium",
                        "High",
                        ...
                    ],
                    "Default": 1, // default quality level index in the Levels array
```

File: content/docs/user-guide/testing/quality/_index.md (hunk 4 above)
Comment (AMZN-alexpete) [reply content/docs/user-guide/testing/quality/_index.md:5]: I'll go with a new section  👍

**Review by lumberyard-employee-dm — APPROVED** [review]
